packages/relay-experimental/FragmentResource.js

```diff
@@ -437,7 +437,7 @@ function createFragmentResource(environment: IEnvironment): FragmentResource {
   return new FragmentResourceImpl(environment);
 }
 
-const dataResources: Map<IEnvironment, FragmentResource> = new Map();
+const dataResources: WeakMap<IEnvironment, FragmentResource> = new WeakMap();
```

Comment:
We need to add here a similar check: 

``` const WEAKMAP_SUPPORTED = typeof WeakMap === 'function';```

Reply:
Wanted too - just wasn't sure how to articulate that to flow?

Reply:
Oh, yeah... I forgot about flow. We probably need to change it to something  ```const dataResources: IMap< IEnvironment, FragmentResource>``` where both WeakMap and Map are implementing this interface IMap.